cfme/base/ui.py

```diff
@@ -1173,11 +1173,51 @@ def is_displayed(self):
         )
 
 
+class SambaProtocolEntities(View):
+    """ Samba Protocol fields on the shedule configuration page """
+    samba_username = Input(id='log_userid')
+    samba_password = Input(id='log_password')
+    samba_confirm_password = Input(id='log_verify')
+
+
+class AWSS3ProtocolEntities(View):
+    """ AWS S3 Protocol fields on the shedule configuration page"""
+    aws_region = BootstrapSelect(id='log_aws_region')
+    aws_username = Input(id='log_userid')
+    aws_password = Input(id='log_password')
+    aws_confirm_password = Input(id='log_verify')
+
+
+class OpenstackSwiftProtocolEntities(View):
+    """ Openstack Swift Protocol fields on the shedule configuration page"""
+    openstack_keystone_version = BootstrapSelect(id='keystone_api_version')
+    openstack_region = Input('openstack_region')
+    openstack_security_protocol = BootstrapSelect(id='security_protocol')
+    openstack_api_port = Input('swift_api_port')
+    openstack_username = Input(id='log_userid')
+    openstack_password = Input(id='log_password')
+    openstack_confirm_password = Input(id='log_verify')
+
+
+class RegionDiagnosticsDatabaseBackupEntities(View):
+    """ Database Backup fields on the shedule configuration page """
+    backup_type = BootstrapSelect('log_protocol')
```

Comment:
I would like to see this implemented as a conditional view, because this type selection controls display of the samba/aws/openstack protocol forms.

Going to consider this optional for this PR, but would be a better way to model the page.

Reply:
Reworked.